src/ORM/Event/TableEvent.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.1
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Event;
+
+use Cake\Controller\Controller;
+use Cake\Event\Event;
+use Cake\ORM\Table;
+
+/**
+ * {@inheritdoc}
+ */
+class TableEvent extends Event
+{
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param string $name Name of the event
+     * @param \Cake\ORM\Table $controller the table that this event applies to
+     * @param array|null $data any value you wish to be transported with this event to it can be read by listeners
+     */
+    public function __construct($name, Table $table, $data = null)
```

Comment:
Does this mean I'll have to implement it in other places as well other then the ```Table``` class?